tests/structure/test_charges.py

```diff
@@ -0,0 +1,308 @@
+    """
+    charges_ = charges.partial_charges(molecule)
+    assert charges_[molecule.element == "C"].tolist() == \
+        pytest.approx(expected_results, abs=1e-2)
+
+
+@pytest.mark.parametrize("molecule", [
+    methane,
+    ethane,
+    ethylene,
+    acetylene,
+    fluoromethane,
+    difluoromethane,
+    trifluoromethane,
+    tetrafluoromethane,
+    fluoroethane,
+    trifluoroethane,
+    methanole,
+    dimethyl_ether,
+    formaldehyde,
+    acetaldehyde,
+    acetone,
+    hydrogen_cyanide,
+    acetonitrile
+])
+def test_total_charge_zero(molecule):
+    """
+    In the case of the 17 molecules given in table 3, it is verified
+    whether the sum of all partial charges equals the sum
+    of all formal charges (in our case zero since we are exclusively
+    dealing with uncharged molecules).
+    """
+    total_charge = np.sum(charges.partial_charges(molecule))
+    assert total_charge == pytest.approx(0, abs=1e-15)
+
+
+# Test whether a formal charge of +1 has the expected effects on
+# carbons's partial charge in methane
+# AtomArray with positive carbon must be created ab inition in order to
+# prevent influencing 'methane'
+pos_methane = array(
+    [carbon, hydrogen, hydrogen, hydrogen, hydrogen]
+)
+pos_methane.bonds = BondList(
+    pos_methane.array_length(),
+    np.array([[0,1], [0,2], [0,3], [0,4]])
+)
+pos_methane.charge = np.array([1, 0, 0, 0, 0])
+
+def test_pos_formal_charge():
+    """
+    Test whether the partial charge of carbon in methane behaves as
+    expected if it carries a formal charge of +1. To be more precise,
+    it is expected to be smaller than 1 since this is the value which
+    negative charge is addded to during iteration and also greater than
+    the partial charge of carbon in methane carrying no formal charge.
+    """
+    ref_carb_part_charge = charges.partial_charges(
+        methane, iteration_step_num=6
+    )[0]
+    pos_carb_part_charge = charges.partial_charges(
+        pos_methane, iteration_step_num=6
+    )[0]
+    assert pos_carb_part_charge < 1
+    assert pos_carb_part_charge > ref_carb_part_charge
+
+
+def test_valence_state_not_parametrized():
+    """
+    Test case in which parameters for a certain valence state of a
+    generally parametrized atom are not available.
+    In our case, it is sulfur having a double bond, i. e. only one
+    binding partner.
+    For this purpose, a fictitious molecule consisting of a central
+    carbon bound to two hydrogen atoms via single bonds and to one
+    sulfur atom via a double bond is created and tested.
+    The expectations are the following: the sulfur's partial charge to
+    be NaN and the carbons's partial charge to be smaller than that of
+    the two hydrogens.
+    """
+    fictitious_molecule = array(
+        [carbon, sulfur, hydrogen, hydrogen]
+    )
+    fictitious_molecule.bonds = BondList(
+        fictitious_molecule.array_length(),
+        np.array([[0,1], [0,2], [0,3]])
+    )
```

Comment:
Please add the `charge` annotation to the `molecule` to suppress the raised warning about missing charges. I also think the warning about the missing parametrization should be asserted with `pytest.warns()`. See https://docs.pytest.org/en/3.0.5/recwarn.html for reference.

Reply:
Done.